tests/MemcacheTest.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+namespace TraderInteractive\Memoize;
+
+use PHPUnit\Framework\TestCase;
+
+/**
+ * @coversDefaultClass \TraderInteractive\Memoize\Memcache
+ * @covers ::<private>
+ */
+class MemcacheTest extends TestCase
+{
+    /**
+     * @test
+     * @covers ::__construct
+     * @covers ::memoizeCallable
+     */
+    public function memoizeCallableWithCachedValue()
+    {
+        $count = 0;
+        $key = 'foo';
+        $value = 'bar';
+        $cachedValue = json_encode(['result' => $value]);
+        $compute = function () use (&$count, $value) {
+            $count++;
+
+            return $value;
+        };
+
+
+        $client = $this->getMemcacheMock();
+        $client->expects(
+            $this->once()
+        )->method('get')->with($this->equalTo($key))->will($this->returnValue($cachedValue));
+
+        $memoizer = new Memcache($client);
+
+        $this->assertSame($value, $memoizer->memoizeCallable($key, $compute));
+        $this->assertSame(0, $count);
+    }
+
+    /**
+     * @test
+     * @covers ::__construct
+     * @covers ::memoizeCallable
+     */
+    public function memoizeCallableWithExceptionOnGet()
+    {
+        $count = 0;
+        $key = 'foo';
+        $value = 'bar';
+        $compute = function () use (&$count, $value) {
+            $count++;
+
+            return $value;
+        };
+
+        $client = $this->getMemcacheMock();
+        $client->expects(
+            $this->once()
+        )->method('get')->with($this->equalTo($key))->will($this->throwException(new \Exception()));
+
+        $memoizer = new Memcache($client);
+
+        $this->assertSame($value, $memoizer->memoizeCallable($key, $compute));
+        $this->assertSame(1, $count);
+    }
+
+    /**
+     * @test
+     * @covers ::__construct
+     * @covers ::memoizeCallable
+     */
+    public function memoizeCallableWithUncachedKey()
+    {
+        $count = 0;
+        $key = 'foo';
+        $value = 'bar';
+        $cachedValue = json_encode(['result' => $value]);
+        $cacheTime = 1234;
+        $compute = function () use (&$count, $value) {
+            $count++;
+
+            return $value;
+        };
+
+        $client = $this->getMemcacheMock();
+        $client->expects($this->once())->method('get')->with($this->equalTo($key))->will($this->returnValue(null));
+        $client->expects($this->once())->method('set')->with($this->equalTo($key), $this->equalTo($cachedValue), $this->equalTo(0), $this->equalTo($cacheTime));
```

Comment:
Break line to address code code enforced standards:
```suggestion
        $client->expects($this->once())->method('set')
            ->with($this->equalTo($key), $this->equalTo($cachedValue), $this->equalTo(0), $this->equalTo($cacheTime));
```